src/CoreBundle/private/library/classes/TCMSFields/TCMSFieldLookupMultiselectCheckboxes.class.php

```diff
@@ -232,9 +233,17 @@ protected function GetMLTFilterQuery()
             $oTableList->sRestriction = null; // do not include the restriction - it is part of the parent table, not the mlt!
 
             $sFilterQuery = $oTableList->FilterQuery().$this->GetMLTRecordRestrictions();
-            $sFilterQueryOrderInfo = $oTableList->GetSortInfoAsString();
-            if (!empty($sFilterQueryOrderInfo)) {
-                $sFilterQuery .= ' ORDER BY '.$sFilterQueryOrderInfo;
+//            $sFilterQueryOrderInfo = $oTableList->GetSortInfoAsString();
+//            if (!empty($sFilterQueryOrderInfo)) {
+//                $sFilterQuery .= ' ORDER BY '.$sFilterQueryOrderInfo;
+//            }
+
+            // TODO check for tabname and cms_field_conf ?? restrict special case to cms_field_conf?
+            // TODO also how does this relate to $bShowCustomsort above?
+
+            if (false !== \strpos($sFilterQuery, 'AS tabname') && false !== \strpos($sFilterQuery, 'cms_field_conf')) {
+                // TODO multilanguage sort! Also for tabname!
+                $sFilterQuery .= ' ORDER BY tabname ASC, `cms_field_conf`.`translation` ASC ';
```

Comment:
I think $bShowCustomsort needs to force a sort by a position field, because in this mode the user changes the positions of a record by a sort order field. This can't work with additional sorting on top.

Reply:
(also see above)

Isn't "custom sort order" a topic for the actual data of a table?
But in this case ("show fields from table") we are only talking about fields and not the table data?

Reply:
I am not sure however as for the general name of the methods (FetchMLTRecords() -> GetMLTFilterQuery()) if this location has to serve two very different purposes and thus cannot work for both.